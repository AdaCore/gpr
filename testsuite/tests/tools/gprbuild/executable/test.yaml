description: Compile an executable with gprbuild
driver: python_script
