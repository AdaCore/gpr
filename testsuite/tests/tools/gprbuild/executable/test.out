================================================================
Case 1 - Build from scratch
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : '0', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 2 - Missing main.o
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 3 - Modified main.o (comments only)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 4 - Missing main.ali
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 5 - Modified main.ali (comments only)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 6 - Missing pkg.o
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 7 - Modified pkg.o (comments only)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 8 - Missing pkg.ali
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 9 - Modified pkg.ali (comments only)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 10 - Missing dep_two.o
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 11 - Modified dep_two.o (comments only)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 12 - Missing dep_two.ali
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 13 - Modified dep_two.ali (comments only)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 14 - Missing b__main.o
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 15 - Modified b__main.o (comments only)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 16 - Missing b__main.ali
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 17 - Modified b__main.ali (comments only)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 18 - Missing main
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : '0', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 19 - Modified main (comments only)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : '0', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 20 - Missing main.adb
main.gpr:5:18: error: main.adb is not a source of project Main
gprbuild: Failed to update sources
Test return value: 4

================================================================
Case 21 - Modified main.adb (comments only)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 22 - Missing pkg.adb
Job 'Compile Ada: PKG (main.gpr)' failed. Status: 1, output: '', stderr: 'cannot generate code for file pkg.ads (package spec)
'
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : '1', stdout: '', stderr: 'cannot generate code for file pkg.ads (package spec)
'
Error detected in jobs.json

================================================================
Case 23 - Modified pkg.adb (comments only)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 24 - Missing pkg.ads
Job 'Compile Ada: MAIN (main.gpr)' failed. Status: 1, output: '', stderr: 'main.adb:1:06: error: "Main (body)" depends on "Pkg (body)"
main.adb:1:06: error: "Pkg (body)" depends on "Pkg (spec)"
main.adb:1:06: error: file "pkg.ads" not found
'
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : '1', stdout: '', stderr: 'main.adb:1:06: error: "Main (body)" depends on "Pkg (body)"
main.adb:1:06: error: "Pkg (body)" depends on "Pkg (spec)"
main.adb:1:06: error: file "pkg.ads" not found
'
Error detected in jobs.json

================================================================
Case 25 - Modified pkg.ads (comments only)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 26 - Missing dep_two.adb
Job 'Compile Ada: DEP_TWO (main.gpr)' failed. Status: 1, output: '', stderr: 'cannot generate code for file dep_two.ads (package spec)
'
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : '1', stdout: '', stderr: 'cannot generate code for file dep_two.ads (package spec)
'
Error detected in jobs.json

================================================================
Case 27 - Modified dep_two.adb (comments only)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 28 - Missing dep_two.ads
Job 'Compile Ada: PKG (main.gpr)' failed. Status: 1, output: '', stderr: 'pkg.adb:2:06: error: "Pkg (body)" depends on "Dep_Two (body)"
pkg.adb:2:06: error: "Dep_Two (body)" depends on "Dep_Two (spec)"
pkg.adb:2:06: error: file "dep_two.ads" not found
'
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : '1', stdout: '', stderr: 'pkg.adb:2:06: error: "Pkg (body)" depends on "Dep_Two (body)"
pkg.adb:2:06: error: "Dep_Two (body)" depends on "Dep_Two (spec)"
pkg.adb:2:06: error: file "dep_two.ads" not found
'
Error detected in jobs.json

================================================================
Case 29 - Modified dep_two.ads (comments only)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 30 - Missing b__main.ads
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 31 - Modified b__main.ads (comments only)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 32 - Missing b__main.adb
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 33 - Modified b__main.adb (comments only)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 34 - Modified main.adb with different return code
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : '0', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from Dep_Two

================================================================
Case 35 - Modified pkg.adb without dep_two dependency
Job 'Link: main (main.gpr)' failed. Status: 1, output: '', stderr: '<gcc>/bin/../lib/gcc/(host)/(gcc-version)/../../../../(host)/bin/ld: /tree/obj/b__main.o: in function `adainit':
b__main.adb:(.text+0x411): undefined reference to `dep_two_E'
<gcc>/bin/../lib/gcc/(host)/(gcc-version)/../../../../(host)/bin/ld: b__main.adb:(.text+0x41b): undefined reference to `dep_two_E'
collect2: error: ld returned 1 exit status
'
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : '1', stdout: '', stderr: '<gcc>/bin/../lib/gcc/(host)/(gcc-version)/../../../../(host)/bin/ld: /tree/obj/b__main.o: in function `adainit':
b__main.adb:(.text+0x411): undefined reference to `dep_two_E'
<gcc>/bin/../lib/gcc/(host)/(gcc-version)/../../../../(host)/bin/ld: b__main.adb:(.text+0x41b): undefined reference to `dep_two_E'
collect2: error: ld returned 1 exit status
'
Error detected in jobs.json

================================================================
Case 36 - Modified dep_two.adb (new output)
== Content of tree/obj/jobs.json:
uid: 'Compile Ada: MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: PKG (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada: DEP_TWO (main.gpr)', status : '0', stdout: '', stderr: ''
uid: 'Bind Ada: main (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Compile Ada post-bind: ADA_MAIN (main.gpr)', status : 'SKIPPED', stdout: '', stderr: ''
uid: 'Link: main (main.gpr)', status : '0', stdout: '', stderr: ''
== Executable output:
Hello from Pkg
Hello from modified Dep_Two

